fix(samples): net: sockets: dumb_http_server: Fix build for native_sim

seyoungjeong opened the pull request:
The dumb_http_server sample was failing to build for native_sim because the sample.yaml file had a filter that excluded platforms with CONFIG_NATIVE_LIBC. The native_sim board uses the native C library, so this filter prevented the sample from being built.

This commit fixes the issue by modifying the filter to allow the sample to be built for native_sim.

jukkar reviewed (commented):
Please change the commit title to
`samples: net: sockets: dumb_http_server: Fix build for native_sim`

The commit is missing Signed-off-by

seyoungjeong committed:
samples: net: sockets: dumb_http_server: Fix build for native_sim

The dumb_http_server sample was failing to build for native_sim
because the sample.yaml file had a filter that excluded platforms
with CONFIG_NATIVE_LIBC. The native_sim board uses the native C
library, so this filter prevented the sample from being built.

Signed-off-by: Seyoung Jeong <seyoungjeong@gmail.com>

aescolar reviewed (changes requested):
Thanks for your contribution @seyoungjeong .
Unfortunately this is not correct:

* This sample builds and runs for native_sim as is.
* Native sim can be built with a choice of C libraries. When this sample is built it defaults to build with picolibc, not the host C library (`CONFIG_NATIVE_LIBC`)
* Even if it were built with `CONFIG_NATIVE_LIBC`, the filter as it is right now would already prevent it from trying to build beyond the configuration step. The added `CONFIG_BOARD..` filter would not change how much is built.

seyoungjeong committed:
Revert "samples: net: sockets: dumb_http_server: Fix build for native_sim"

This reverts commit 9256f68690d1fcb080310a1ae67e3bab4cc97424.

The previous change was incorrect because native_sim can be built with different C libraries, and the sample already builds correctly with picolibc. The filter was not necessary.

Signed-off-by: Seyoung Jeong <seyoungjeong@gmail.com>

seyoungjeong commented:
Thanks @aescolar I've closed the pull request.
